docker/archive/dest.go

```diff
@@ -19,17 +19,32 @@ func newImageDestination(ctx *types.SystemContext, ref archiveReference) (types.
 	if ref.destinationRef == nil {
 		return nil, errors.Errorf("docker-archive: destination reference not supplied (must be of form <path>:<reference:tag>)")
 	}
-	fh, err := os.OpenFile(ref.path, os.O_WRONLY|os.O_EXCL|os.O_CREATE, 0644)
+	// tries to open a file first to accomodate for load when using stdout
+	// as the file is created first then written to
+	// if the file doesn't exist then it creates a new file for exclusive use
+	// if the file exists, it checks to ensure the size is 0
```

Comment:
> append to the tar ball soon

You mean to update an archive by adding new files to the end of an existing tarball? That works with the “tar as tape processed at once” model (and seems to be documented at https://www.gnu.org/software/tar/manual/html_node/multiple.html ), but it's not obviously interoperable (in particular, the reader in this package stops at the first match). Of course that particular reader can be updated, but in general?

Ultimately, what is the use case here? Why would anyone want to keep updating a set of images in a giant single file with O(N) seeking and no way to delete a layer in the middle, as opposed to using either a registry or the OCI expanded dirs+files format?